### PR TITLE
Use Dune

### PR DESCRIPTION
**Pull request by samoht**

### CODE REVIEW

**Comment by samoht** [comment]
/ping @djs55 and @avsm (and @emillon :p)

File: examples/dune [hunk 1]
```diff
@@ -0,0 +1,9 @@
+(executables
+  (names     forward server)
+  (libraries dns-lwt-unix))
+
+(alias
+  (name    runtest)
+  (deps    forward.exe server.exe)
+; (action  (run ${<} -e -v))
```
Comment (emillon) [inline examples/dune]: you can probably convert that to (tests), I think it supports (deps).

File: lib_test/ounit/dune [hunk 2]
```diff
@@ -0,0 +1,9 @@
+(executable
+  (name       test)
+  (libraries  dns ounit pcap-format)
+  (preprocess (pps ppx_cstruct)))
+
+(alias
+  (name    runtest)
+  (deps    test.exe (glob_files *.pcap) (glob_files *.zone))
+  (action  (run ${<})))
```
Comment (emillon) [inline lib_test/ounit/dune]: In a dune file, that probably tries to run the wrong thing.

Comment (samoht) [reply lib_test/ounit/dune]: ha yes, good catch

File: Makefile [hunk 3]
```diff
@@ -1,32 +1,10 @@
...
-# until we have https://github.com/ocaml/opam-publish/issues/38
-pkg-%:
-	topkg opam pkg -n $*
-	mkdir -p $(PACKAGES)/$*
-	cp -r _build/$*.* $(PACKAGES)/$*/
-	cd $(PACKAGES) && git add $*
-
-PKGS=$(basename $(wildcard *.opam))
-opam-pkg:
-	$(MAKE) $(PKGS:%=pkg-%)
```
Comment (djs55) [inline Makefile]: Out of curiosity, what's the recommended way of creating+publishing packages?

File: examples/dune [hunk 4]
```diff
@@ -0,0 +1,9 @@
+(executables
+  (names     forward server)
+  (libraries dns-lwt-unix))
+
+(alias
+  (name    runtest)
+  (deps    forward.exe server.exe)
+; (action  (run ./forward.exe -e -v))
```
Comment (djs55) [inline examples/dune]: Is the `;` commenting-out the action?

File: lib_test/unix/dune [hunk 5]
```diff
@@ -0,0 +1,9 @@
+(executable
+  (name      time_server)
+  (libraries dns-lwt-unix))
+
+(alias
+  (name   runtest)
+  (deps   time_server.exe)
+;  (action (run %{deps} --help))
```
Comment (djs55) [inline lib_test/unix/dune]: If this is supposed to be commented-out maybe we should add a comment explaining why.

File: Makefile (hunk 3 above)
Comment (samoht) [reply Makefile]: you can use [dune-release](https://github.com/samoht/dune-release). That's not perfect but does relatively well in general :-) Feel free to open issues if that's not the case.

File: examples/dune (hunk 4 above)
Comment (samoht) [reply examples/dune]: yes, it was like that before; I guess it's because that test needs the server to run first?

Comment (djs55) [reply examples/dune]: Ah fair enough, I didn't realise it was already there

**Pull request merged**